src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -410,15 +413,16 @@ private synchronized int execMatlabCommand(Run<?, ?> build, FilePath workspace,
             throws IOException, InterruptedException {
```

Comment:
Let's try to reduce the level of detail in this top level function. This function should read like the business logic of our plugin implementation

1. Determine the platform on which we will run (Input: perform method)
2. Determine the channel
3. Determine the MATLAB root on the remote machine (Input: Output of 2, class property)
4. Determine the MATLAB version (Inputs : output of 3)
5. Construct the MATLAB command (Inputs: Output of 4 and output of 1)
6. Determine the node workspace
7. Invoke the MATLAB command (Inputs: Output of 5 and 6)

Reply:
Done. refactored the code as close as possible to read it in the suggested way.